src/scripting/lua_map.cc

```diff
@@ -8090,9 +8098,8 @@ int LuaField::get_claimers(lua_State* L) {
 
 	std::vector<PlrInfluence> claimers;
 
-	iterate_players_existing(other_p, map.get_nrplayers(), egbase, plr)
-	   claimers.push_back(PlrInfluence(
-	      plr->player_number(), plr->military_influence(map.get_index(coords_, map.get_width()))));
+	iterate_players_existing(other_p, map.get_nrplayers(), egbase, plr) claimers.emplace_back(
+	   plr->player_number(), plr->military_influence(map.get_index(coords_, map.get_width())));
```

Comment:
Although the change is ok against the previous state, I amn confused by this line as it seems a ; might have been missing, but in this case the compiler shpould have complained. Could you explain why there is none before the modification of the claimers vector?

Reply:
`iterate_players_existing` is a macro that expands to a `for` loop, so statement this is a shorthand for
```c++
for (Widelands::PlayerNumber other_p = 1; other_p < map.get_nrplayers() + 1; ++other_p)
	if (Widelands::Player* const plr = (egbase).get_player(other_p))
		claimers.emplace_back(...);
```
And yes those `iterate_...` macros are horrible for readability…

Reply:
ok thanks